src/poseidon/mod.rs

```diff
@@ -14,10 +13,30 @@ pub mod constraints;
 #[cfg(test)]
 mod tests;
 
-#[derive(Clone)]
-enum PoseidonSpongeMode {
-    Absorbing { next_absorb_index: usize },
-    Squeezing { next_squeeze_index: usize },
+/// default parameters traits for Poseidon
+pub mod traits;
+pub use traits::*;
+
+mod grain_lfsr;
+
+/// Parameters and RNG used
+#[derive(Clone, Debug)]
+pub struct PoseidonParameters<F: PrimeField> {
+    /// number of rounds in a full-round operation
+    pub full_rounds: usize,
+    /// number of rounds in a partial-round operation
+    pub partial_rounds: usize,
+    /// Exponent used in S-boxes
+    pub alpha: u64,
+    /// Additive Round keys. These are added before each MDS matrix application to make it an affine shift.
+    /// They are indexed by `ark[round_num][state_element_index]`
+    pub ark: Vec<Vec<F>>,
+    /// Maximally Distance Separating Matrix.
+    pub mds: Vec<Vec<F>>,
+    /// the rate (in terms of number of field elements)
+    pub rate: usize,
+    /// the capacity (in terms of number of field elements)
```

Comment:
```suggestion
    /// The rate (in terms of number of field elements)
    pub rate: usize,
    /// The capacity (in terms of number of field elements)
```

Also, could you add a link to, or an inline explanation of, what these terms mean?